src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionExpirationQueueConsumerTests.kt

```diff
@@ -3258,4 +3282,333 @@ class TransactionExpirationQueueConsumerTests {
+          transactionAuthorizationRequestedEvent(
+            TransactionAuthorizationRequestData.PaymentGateway.NPG,
+            NpgTransactionGatewayAuthorizationRequestedData(
+              URI.create("http://localhost"),
+              "MC",
+              UUID.randomUUID().toString(),
+              UUID.randomUUID().toString())),
+          transactionExpiredEvent(TransactionStatusDto.EXPIRED))
+      setupTransactionStorageMock(events)
+      given(checkpointer.success()).willReturn(Mono.empty())
+      given { authorizationStateRetrieverService.getStateNpg(any(), any(), any(), any(), any()) }
+        .willAnswer { Mono.error<StateResponseDto>(NpgServerErrorException("Any 5xx error")) }
+
+      given { refundRetryService.enqueueRetryEvent(any(), any(), any()) }.willAnswer { mono {} }
+
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.right(
+            QueueEvent(
+              events.filterIsInstance<TransactionExpiredEvent>().first(), MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsRefundedEventStoreRepository, times(2)).save(any())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundRequestedEvent>()
+          .count())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundErrorEvent>()
+          .count())
+      verify(deadLetterTracedQueueAsyncClient, times(0))
+        .sendAndTraceDeadLetterQueueEvent(any(), any())
+      verify(refundRetryService, times(1)).enqueueRetryEvent(any(), any(), any())
+    }
+  }
+
+  private fun setupTransactionStorageMock(events: List<TransactionEvent<*>>) {
+    val transaction = reduceEvents(events.map { it as TransactionEvent<Any> }.toFlux()).block()
+    given(
+        transactionsEventStoreRepository.findByTransactionIdOrderByCreationDateAsc(
+          any(),
+        ))
+      .willReturn(events.map { it as TransactionEvent<Any> }.toFlux())
+    given(
+        transactionsExpiredEventStoreRepository.save(transactionExpiredEventStoreCaptor.capture()))
+      .willAnswer { Mono.just(it.arguments[0]) }
+    given(
+        transactionsRefundedEventStoreRepository.save(transactionRefundEventStoreCaptor.capture()))
+      .willAnswer { Mono.just(it.arguments[0]) }
+    given(transactionsViewRepository.save(transactionViewRepositoryCaptor.capture())).willAnswer {
+      Mono.just(it.arguments[0])
+    }
+    given(transactionsViewRepository.findByTransactionId(TRANSACTION_ID)).willAnswer {
+      transactionDocument(transaction!!.status, ZonedDateTime.now()).toMono()
+    }
+  }
+
+  companion object {
+    @JvmStatic
+    fun manualCheckRequiredNPGResponses(): Stream<Arguments> =
+      Stream.of<Either<StateResponseDto, Exception>>(
```

Comment:
🚲
flip Either left and right value so that exception is mapped to left instead of right